pkg/api/api.go

```diff
@@ -194,9 +195,49 @@ func (s *server) newTracingHandler(spanName string) func(h http.Handler) http.Ha
 	}
 }
 
+// checkOrigin returns true if the origin is not set or is equal to the request host.
+func (s *server) checkOrigin(r *http.Request) bool {
+	origin := r.Header["Origin"]
+	if len(origin) == 0 {
+		return true
+	}
+	hosts := append(s.CORSAllowedOrigins, "http://"+r.Host)
+	for _, v := range hosts {
+		if equalASCIIFold(origin[0], v) || v == "*" {
+			return true
+		}
+	}
+
+	return false
+}
+
 func lookaheadBufferSize(size int64) int {
 	if size <= largeBufferFilesizeThreshold {
 		return smallFileBufferSize
 	}
 	return largeFileBufferSize
 }
+
+// equalASCIIFold returns true if s is equal to t with ASCII case folding as
```

Comment:
so does this mean one can abuse the cors check by creating a url that asciifold-equal to the ligit one?

Reply:
the ascii fold is to convert to lowercase (that's the IETF standard as per the RFC), and AFAIK hostname is case insensitive